projects/batfish-common-protocol/src/main/java/org/batfish/datamodel/pojo/Environment.java

```diff
@@ -3,52 +3,53 @@
+import java.util.Date;
 import java.util.Objects;
 import javax.annotation.Nullable;
-import org.batfish.datamodel.Edge;
-import org.batfish.datamodel.collections.NodeInterfacePair;
 
 /**
  * The {@link Environment Environment} is an Object representation of the environment for BatFish
  * service.
  *
- * <p>Each {@link Environment Environment} contains a name and all informations of the environment
- * {@link #_name}.
+ * <p>Each {@link Environment Environment} contains a name and a summary of the environment {@link
+ * #_name}.
  */
 public class Environment {
   private static final String PROP_NAME = "name";
-  private static final String PROP_EDGE_BLACKLIST = "edgeBlacklist";
-  private static final String PROP_INTERFACE_BLACKLIST = "interfaceBlacklist";
-  private static final String PROP_NODE_BLACKLIST = "nodeBlacklist";
-  private static final String PROP_ENVIRONMENT_BGP_TABLES = "bgpTables";
-  private static final String PROP_ENVIRONMENT_ROUTING_TABLES = "routingTables";
+  private static final String PROP_EDGE_BLACKLIST_COUNT = "edgeBlacklistCount";
+  private static final String PROP_INTERFACE_BLACKLIST_COUNT = "interfaceBlacklistCount";
+  private static final String PROP_NODE_BLACKLIST_COUNT = "nodeBlacklistCount";
+  private static final String PROP_ENVIRONMENT_BGP_TABLES_COUNT = "bgpTablesCount";
+  private static final String PROP_ENVIRONMENT_ROUTING_TABLES_COUNT = "routingTablesCount";
   private static final String PROP_EXTERNAL_BGP_ANNOUNCEMENTS = "externalBgpAnnouncements";
+  private static final String PROP_CREATED_AT = "createdAt";
 
   private final String _name;
-  private final List<Edge> _edgeBlacklist;
-  private final List<NodeInterfacePair> _interfaceBlacklist;
-  private final List<String> _nodeBlacklist;
-  private final Map<String, String> _bgpTables;
-  private final Map<String, String> _routingTables;
+  private final Date _createdAt;
+  private final int _edgeBlacklistCount;
+  private final int _interfaceBlacklistCount;
+  private final int _nodeBlacklistCount;
+  private final int _bgpTablesCount;
+  private final int _routingTablesCount;
   private final String _externalBgpAnnouncements;
 
   @JsonCreator
   public Environment(
       @JsonProperty(PROP_NAME) String name,
-      @JsonProperty(PROP_EDGE_BLACKLIST) @Nullable List<Edge> edgeBlacklist,
-      @JsonProperty(PROP_INTERFACE_BLACKLIST) @Nullable List<NodeInterfacePair> interfaceBlacklist,
-      @JsonProperty(PROP_NODE_BLACKLIST) @Nullable List<String> nodeBlacklist,
-      @JsonProperty(PROP_ENVIRONMENT_BGP_TABLES) @Nullable Map<String, String> bgpTables,
-      @JsonProperty(PROP_ENVIRONMENT_ROUTING_TABLES) @Nullable Map<String, String> routingTables,
+      @JsonProperty(PROP_CREATED_AT) Date createdAt,
+      @JsonProperty(PROP_EDGE_BLACKLIST_COUNT) int edgeBlacklistCount,
+      @JsonProperty(PROP_INTERFACE_BLACKLIST_COUNT) int interfaceBlacklistCount,
+      @JsonProperty(PROP_NODE_BLACKLIST_COUNT) int nodeBlacklistCount,
```

Comment:
We should have node Blacklist in the object itself as it is just a list of strings and may be loaded with the object in the memory